picoquic/picoquic.h

```diff
@@ -190,6 +192,22 @@ typedef void (*picoquic_stream_data_cb_fn)(picoquic_cnx_t* cnx,
 typedef void (*cnx_id_cb_fn)(picoquic_connection_id_t cnx_id_local,
     picoquic_connection_id_t cnx_id_remote, void* cnx_id_cb_data, picoquic_connection_id_t * cnx_id_returned);
 
+/* Will be called to verify that the given data corresponds to the given signature.
+ * This callback and the `verify_ctx` will be set by the `verify_certificate_cb_fn`.
+ * If `data` and `sign` are empty buffers, an error occurred and `verify_ctx` should be freed.
+ * Expect `0` as return value, when the data matches the signature.
+ */
+typedef int (*verify_sign_cb_fn)(void* verify_ctx, ptls_iovec_t data, ptls_iovec_t sign);
+/* Will be called to verify a certificate of a connection.
+ * The arguments `verify_sign` and `verify_sign_ctx` are expected to be set, when the function returns `0`.
+ * See `verify_sign_cb_fn` for more information about these arguments.
+ */
+typedef int (*verify_certificate_cb_fn)(void* ctx, picoquic_cnx_t* cnx, ptls_iovec_t* certs, size_t num_certs,
+                                        verify_sign_cb_fn* verify_sign, void** verify_sign_ctx);
+
```

Comment:
Why void**?

Reply:
Because this is call by reference and we require the implementation to set a pointer.